internal/app/backend/backend_service.go

```diff
@@ -95,59 +101,64 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 	}
 
 	ctx := stream.Context()
-	matchChan := make(chan *pb.Match)
-	errChan := make(chan error)
+	resultChan := make(chan mmfResult)
 
-	go func(matchChan chan<- *pb.Match, errChan chan<- error) {
+	go func(resultChan chan<- mmfResult) {
 		var wg sync.WaitGroup
+		var matches []*pb.Match
 
 		defer func() {
 			wg.Wait()
-			close(matchChan)
-			close(errChan)
+			close(resultChan)
 		}()
 
 		for _, profile := range req.Profile {
 			wg.Add(1)
 			go func(profile *pb.MatchProfile) {
 				defer wg.Done()
 
-				// Get the channel over which the generated match results will be sent.
+				// Get the match results that will be sent.
+				// TODO: The matches returned by the MatchFunction will be sent to the
+				// Evaluator to select results. Until the evaluator is implemented,
+				// we channel all matches as accepted results.
 				switch (req.Config.Type).(type) {
 				case *pb.FunctionConfig_Grpc:
-					matchesFromGRPCMMF(ctx, profile, grpcClient, matchChan, errChan)
+					matches, err = matchesFromGRPCMMF(ctx, profile, grpcClient)
 				case *pb.FunctionConfig_Rest:
-					matchesFromHTTPMMF(ctx, profile, httpClient, baseURL, matchChan, errChan)
+					matches, err = matchesFromHTTPMMF(ctx, profile, httpClient, baseURL)
 				}
+
+				resultChan <- mmfResult{matches, err}
 			}(profile)
 		}
-	}(matchChan, errChan)
+	}(resultChan)
 
-	for {
+	var done bool
+
+	for result := range resultChan {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
-		case err := <-errChan:
-			return err
-		case match, ok := <-matchChan:
-			if !ok {
-				// Channel closed indicating that we are done processing matches.
-				return nil
+			done = true
+			continue
+		default:
+			// Log and continue with the results from next profile if the current ones fail
+			if result.err != nil {
```

Comment:
because of this logic, it could fail to run the MMF, but return no error, only logging it.  We should be returning an error in this case, right?  Note, we do want to avoid logging errors more than once per system.